apis/r/R/SOMASparseNDArray.R

```diff
@@ -74,7 +78,76 @@ SOMASparseNDArray <- R6::R6Class(
         j = bit64::as.integer64(values@j),
         x = values@x
       )
-      colnames(coo) <- c(self$dimnames(), self$attrnames())
+      dnames <- self$dimnames()
+      colnames(coo) <- c(dnames, self$attrnames())
+      ranges <- sapply(
+        X = dnames,
+        FUN = function(x) {
+          return(range(coo[[x]]))
+        },
+        simplify = FALSE,
+        USE.NAMES = TRUE
+      )
+      bbox <- bbox %||% ranges
+      if (is.null(names(bbox))) {
+        names(bbox) <- dnames
+      }
+      if (!is_named(bbox, allow_empty = FALSE)) {
+        idx <- which(!nzchar(names(bbox)))
+        names(bbox)[idx] <- dnames[idx]
+      }
+      bbox <- bbox[dnames]
+      if (length(bbox) != length(dnames)) {
+        stop("dnames")
+      }
+      if (is_integerish(bbox) || bit64::is.integer64(bbox)) {
+        bbox <- sapply(
+          X = names(bbox),
+          FUN = function(x) {
+            return(sort(c(min(ranges[[x]]), bbox[[x]])))
+          },
+          simplify = FALSE
+        )
+      }
+      for (x in dnames) {
+        xrange <- bbox[[x]]
+        if (any(is.na(xrange))) {
+          stop(
+            "Ranges in the bounding box must be finite (offending: ",
+            sQuote(x),
+            ")",
+            call. = FALSE
+          )
+        }
+        if (!(is_integerish(xrange) || bit64::is.integer64(xrange))) {
+          stop(
+            "Ranges in the bounding box must be integers",
+            call. = FALSE
+          )
+        }
+        xrange <- sort(bit64::as.integer64(xrange))
+        if (length(xrange) != 2L) {
+          stop(
+            "Ranges in the bounding box must be two 64-bit integers",
```

Comment:
must _already_ be, or must be _capable of being converted to_?

Reply:
Capable of being converted to; addressed in f38cc1c